storage/cache/subtree_cache.go

```diff
@@ -197,36 +195,26 @@ func (s *SubtreeCache) cacheSubtree(t *storagepb.SubtreeProto) error {
 
 // GetNodes returns the requested nodes, calling the getSubtrees function if
 // they are not already cached.
-func (s *SubtreeCache) GetNodes(nodeIDs []compact.NodeID, getSubtrees GetSubtreesFunc) ([]tree.Node, error) {
-	// TODO(pavelkalinnikov): Use compact.NodeID directly, don't convert.
-	ids := make([]tree.NodeID, len(nodeIDs))
-	for i, nID := range nodeIDs {
-		id, err := tree.NewNodeIDForTreeCoords(int64(nID.Level), int64(nID.Index), maxSupportedTreeDepth)
-		if err != nil {
-			return nil, fmt.Errorf("failed to create nodeID %v: %v", nID, err)
-		}
-		ids[i] = id
-	}
-
+func (s *SubtreeCache) GetNodes(ids []compact.NodeID, getSubtrees GetSubtreesFunc) ([]tree.Node, error) {
 	glog.V(2).Infof("cache: GetNodes(count=%d)", len(ids))
 	if glog.V(3) {
-		for _, n := range ids {
-			glog.Infof("  cache: GetNodes(path=%x, prefixLen=%d)", n.Path, n.PrefixLenBits)
+		for _, id := range ids {
+			glog.Infof("  cache: GetNodes(%+v)", id)
 		}
 	}
 	if err := s.preload(ids, getSubtrees); err != nil {
 		return nil, err
 	}
 
 	ret := make([]tree.Node, 0, len(ids))
-	for i, id := range ids {
+	for _, id := range ids {
 		h, err := s.getNodeHash(
 			id,
-			func(n tree.NodeID) (*storagepb.SubtreeProto, error) {
+			func(id []byte) (*storagepb.SubtreeProto, error) {
 				// This should never happen - we should've already read all the data we
 				// need above, in Preload()
-				glog.Warningf("Unexpectedly reading from within getNodeHash(): %s", n.String())
-				ret, err := getSubtrees([]tree.NodeID{n})
+				glog.Warningf("Unexpectedly reading from within getNodeHash(): %x", id)
```

Comment:
I think we previously found the above comment to be incorrect. Might demote the log level. This condition is also not tested. Again, I suspect that was true before.

Reply:
I believe it's correct (or don't see why it's not). Also, I found 0 occurrences of this warning in our logs. I think we should rather return an error here, or don't have this callback in the first place. To be addressed in follow-ups.

Reply:
Maybe it was maps again? I remember we tried to remove it before and it did happen.